### PR TITLE
[tests] add a test case of mDNS service restarting

### PR DESCRIPTION
**Pull request by superwhd**
Depends-On: openthread/ot-br-posix#1221

Please merge after openthread/ot-br-posix#1221 is merged.

### CODE REVIEW

File: tests/scripts/thread-cert/border_router/test_mdns_restart.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+        self.assertEqual(len(host.browse_mdns_services('_ed1._tcp')), 1)
+        self.assertEqual(len(host.browse_mdns_services('_ed2._tcp')), 1)
+
+        # BR1's mdns restart
+        br1.bash('service mdns restart')
+        br1.bash('service avahi-daemon restart')
+
+        self.simulator.go(10)
+
+        self.assertEqual(len(host.browse_mdns_services('_meshcop._udp')), 2)
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/border_router/test_mdns_restart.py:151]: should we also test Advertising Proxy ?

Comment (superwhd) [reply tests/scripts/thread-cert/border_router/test_mdns_restart.py:151]: This test case already covers it. After `mdnsd` restarts, ED1 will automatically republish the registered SRP service and ED2 will discover that service via DNSSD. See the last block of code.

File: tests/scripts/thread-cert/border_router/test_mdns_restart.py [hunk 2]
```diff
@@ -0,0 +1,169 @@
+#!/usr/bin/env python3
+#
+#  Copyright (c) 2020, The OpenThread Authors.
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/border_router/test_mdns_restart.py:3]:
```suggestion
#  Copyright (c) 2022, The OpenThread Authors.
```
time flies

File: tests/scripts/thread-cert/border_router/test_mdns_restart.py [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+import logging
+import unittest
+from ipaddress import IPv6Network
+
+import config
+import thread_cert
+
+# Test description:
+#   This test verifies that a single OMR and on-link prefix is chosen
+#   and advertised when there are multiple Border Routers in the same
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/border_router/test_mdns_restart.py:38]: adjust test description accordingly?

**Review by simonlingoogle — APPROVED** [review]
LGTM :+1: 

with nits

**Review by wgtdkp — APPROVED** [review]

**Review by jwhui — APPROVED** [review]

**Pull request merged**